test.js

```diff
@@ -158,6 +154,27 @@ describe('capture-exit', function() {
         exit.onExit(function () { });
       }).to.throw('Cannot install handler when exit is not captured.  Call `captureExit()` first');
     });
+
+    it('throws if an exit is already happening', function() {
```

Comment:
Arguably, we could handle the second script above by changing where we set `isExiting` (because we only _really_ want to prevent the first script (`captureExit.onExit` nested inside `process.on('exit'`) but the second might be ok?).

Moving `isExiting = true` to the `.finally` inside `process.on('beforeExit'` would likely allow the second script (nested `captureExit.onExit`) and disallow the first script (`captureExit.onExit` nested inside `process.on('exit'`)...